### DISCUSSION TITLE
Exception: error.public-api.exceeded-account-historical-data-allowance

### DISCUSSION THREAD

**Discussion by oliverpolden** [ig-python/ig-markets-api-python-library#274] [General]
As of the 3rd Jan 2023 I'm constantly getting this error. I'm running an outdated version of ig-python so I'm curious to know if others are seeing this error now on a recent version. I suspect IG made some sort of release which tightens up request limits. So I'm wondering whether I should spend time debugging my system as it is (might be quick), or spend time updating to the latest version (might take a bit more time). I should probably do the latter anyway.

**Comment by bug-or-feature** [comment]
The recent versions make it a lot easier to [work around the limits](https://trading-ig.readthedocs.io/en/latest/faq.html#how-to-avoid-hitting-the-rate-limits). I only access the API with the Tenacity support now, so I never see the rate limit error

**Reply by oliverpolden** [reply]
Great thank you for your reply. I'll upgrade and give it a go. Just hope I don't have to modify my code too much, it's been a while!

**Reply by starsnpixel** [reply]
> The recent versions make it a lot easier to [work around the limits](https://trading-ig.readthedocs.io/en/latest/faq.html#how-to-avoid-hitting-the-rate-limits). I only access the API with the Tenacity support now, so I never see the rate limit error

I am currently struggling with this solution and I am not sure why because I simply copy-pasted the one from the FAQs, with minor adjustments. I hope you can help me with this. MWE:
```
from trading_ig import IGService, ApiExceededException
from tenacity import Retrying, wait_exponential, retry_if_exception_type
from trading_ig.config import config

def login():
    retryer = Retrying(wait=wait_exponential(),
                       retry=retry_if_exception_type(ApiExceededException))
    ig_service = IGService(config.username, config.password, config.api_key, config.acc_type,
                           acc_number=config.acc_number, retryer=retryer, use_rate_limiter=True)
    ig_service.create_session(version='3')
    print('Connection successfully established')
    return ig_service

session = login()
```

However, PyCharm already marks `from trading_ig import IGService, ApiExceededException` as red and when I try to run it, it says "ImportError: cannot import name 'ApiExceededException' from 'trading_ig'" - any idea why this is? I thought

**Reply by starsnpixel** [reply]
Solved: Solution can be found in https://github.com/ig-python/trading-ig/issues/280#issue-1571466021
